src/main/java/net/dries007/tfc/common/blocks/rock/AqueductBlock.java

```diff
@@ -323,4 +323,41 @@ else if (isValidSource(adjacentState) && getFluidProperty().canContain(adjacentF
             }
         }
     }
+
+    @Override
+    @SuppressWarnings("deprecation")
+    public BlockState rotate(BlockState state, Rotation rot) {
+        return switch (rot) {
+            case CLOCKWISE_90 -> state.setValue(NORTH, state.getValue(WEST))
+                .setValue(EAST, state.getValue(NORTH))
+                .setValue(SOUTH, state.getValue(EAST))
+                .setValue(WEST, state.getValue(SOUTH))
+                .setValue(FACING, rot.rotate(state.getValue(FACING)));
+            case CLOCKWISE_180 -> state.setValue(NORTH, state.getValue(SOUTH))
+                .setValue(EAST, state.getValue(WEST))
+                .setValue(SOUTH, state.getValue(NORTH))
+                .setValue(WEST, state.getValue(EAST))
+                .setValue(FACING, rot.rotate(state.getValue(FACING)));
+            case COUNTERCLOCKWISE_90 -> state.setValue(NORTH, state.getValue(EAST))
+                .setValue(EAST, state.getValue(SOUTH))
+                .setValue(SOUTH, state.getValue(WEST))
+                .setValue(WEST, state.getValue(NORTH))
+                .setValue(FACING, rot.rotate(state.getValue(FACING)));
+            default -> state;
+        };
+    }
+
+    @Override
+    @SuppressWarnings("deprecation")
+    public BlockState mirror(BlockState state, Mirror mirror) {
+        return switch (mirror) {
+            case LEFT_RIGHT -> state.setValue(NORTH, state.getValue(SOUTH))
+                .setValue(SOUTH, state.getValue(NORTH))
+                .setValue(FACING, mirror.mirror(state.getValue(FACING)));
+            case FRONT_BACK -> state.setValue(EAST, state.getValue(WEST))
+                .setValue(WEST, state.getValue(EAST))
+                .setValue(FACING, mirror.mirror(state.getValue(FACING)));
+            default -> super.mirror(state, mirror);
+        };
+    }
```

Comment:
This could be done with two calls - one to a static method in `DirectionPropertyBlock` and one to `super`.